src/engines/html5/media-source/base-media-source-adapter.js

```diff
@@ -186,11 +187,13 @@ export default class BaseMediaSourceAdapter extends FakeEventTarget implements I
    * @private
    */
   _handleLiveTimeUpdate(): void {
-    this._videoElement.addEventListener(Html5EventType.DURATION_CHANGE, () => {
-      if (this.isLive() && this._videoElement.paused) {
-        this._trigger(Html5EventType.TIME_UPDATE);
-      }
-    });
+    this._videoElement.addEventListener(Html5EventType.DURATION_CHANGE, this._onDurationChanged.bind(this));
```

Comment:
Why not use event manager also here?

Reply:
If media source adapter will do remove all listener this listener will be deleted and nobody will re-register it. 
I think we want to make sure it'll not happened

Reply:
this won't remove the listener, you need to keep reference to the bounded method and use that in the add and remove